examples/pip_parse_vendored/WORKSPACE

```diff
@@ -14,14 +14,15 @@ python_register_toolchains(
     python_version = "3.9",
 )
 
-load("@python39//:defs.bzl", "interpreter")
 load("@rules_python//python:pip.bzl", "pip_parse")
 
 # This repository isn't referenced, except by our test that asserts the requirements.bzl is updated.
 # It also wouldn't be needed by users of this ruleset.
 pip_parse(
     name = "pip",
-    python_interpreter_target = interpreter,
+    # Make the requirements.bzl platform agnostic by specifying the
+    # multi-platform toolchain repository.
+    interpreter_repo = "@python39",
```

Comment:
So in the example there was a target `make_platform_agnostic`, which was `sed`ing and adding the import and setting the `interpreter` to something else than a platform-specific label.

When you run `pip_parse` on a particular platform, the `interpreter` value coming from a hermetic `python39` toolchain repo will vary based on which platform we are running on, hence the `python_interpreter_target` value will differ in the resulting file as well. There are two ways to make the file platform-agnostic:
1. Do not store the value at all and force the user to pass `python_interpreter_target` to the `install_deps` call instead.
2. Use this trick of search-and-replace and import the alias repo to abstract away the platform differences.

The previous code chose number 2. because we were using the interpreter to run the `lock_file_generator` tool, however with this refactor, we could in theory go for option 1. as well. I'd be happy to make the `python_interpreter_target` a parameter of the `whl_library` only as a followup PR if there is agreement. For what it is worth, the `interpreter_repo` attribute was an addition to the API and not a backwards incompatible change, but your question here was spot on, thank you.

Reply:
Thanks. I dug into this code a bit and have a better understand of how it works.

I'm inclined to say to keep passing the interpreter to pip_parse instead of install_deps (they seem closely coupled).

I'm not sure if passing the repo name in vs the specific host target is a big win. Hiding that load(alias-repo, interpreter) line does sound nice...but it adds a _third_ way to get the interpreter into pip_parse, which starts to feel a bit bloaty. Like there's some other abstraction we're missing (it reminds me of all the --python_path, --python_top flags, which were ultimately simplified to just py_runtime()).

Anyways, yeah, we can discuss off the PR.